File: 0049-group-anagrams/0049-group-anagrams.dart
class Solution {
  List<List<String>> groupAnagrams(List<String> strs) {
    Map<int, String> map = {};
    int size = strs.length;

    for(int i=0;i<size;i++){
        map[i] = (strs[i].split('')..sort()).join('');
    }
    List<List<String>> res = [];
    List<String> temp = [];
    for(int i=0;i<size;i++){
        var str = (strs[i].split('')..sort()).join('');
        if(map.containsValue(str)){
            var indexList = map.entries
                .where((entry) => entry.value == str)
                .map((entry) => entry.key)
                .toList();
            for(int index in indexList){
                temp.add(strs[index]);
                map.remove(index);
            }
            res.add(List.of(temp));
            temp = [];
        }
    }
    return res;
  }
}

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> map;
        for(int i=0;i<strs.size();i++){
            string order = strs[i];
            sort(order.begin(), order.end());
            map[order].push_back(strs[i]);
        }
        vector<vector<string>> res;
        int index = 0;
        for(auto el : map){
            vector<string> temp;
            for(int i=0;i<el.second.size();i++){
                temp.push_back(el.second[i]);
            }
            res.push_back(temp);
        }
        return res;
    }
};